File: string/14.c
//利用strs[0]做基准, 与之后的string进行比较, 发生不匹配就跳出
char* longestCommonPrefix(char** strs, int strsSize) {
    if(strsSize == 0 || strlen(strs[0]) == 0) return "";
    if(strsSize == 1) return strs[0];
    int prefixindex = 0;
    int flag = true;
    while(flag)
    {
        for(int i = 1; i < strsSize; i++)
        {
            if(strs[0][prefixindex] == strs[i][prefixindex]) continue;
            else
            {
                flag = false;
                break;
            }
        }
        prefixindex++;
    }
    
    if(prefixindex == 0) return "";
    else
    {
        char* ret = malloc(sizeof(char)*(prefixindex));
        strncpy(ret, strs[0], prefixindex - 1);
        ret[prefixindex - 1] = '\0';
        return ret;
    }
}

File: string/14.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    var ret string
    var index int
    for index < len(strs[0]) {
        for i := 1; i < len(strs); i++ {
            if len(strs[i]) <= index || strs[0][index] != strs[i][index] {
                return ret
            }
        }
        ret += string(strs[0][index])
        index++
    }
    return ret
}
